service/api.go

```diff
@@ -460,15 +506,96 @@ func (s *gnoNativeService) Run(ctx context.Context, req *connect.Request[api_gen
 		})
 	}
 
-	bres, err := s.client.Run(cfg, msgs...)
+	return cfg, msgs
+}
+
+func (s *gnoNativeService) MakeCallTx(ctx context.Context, req *connect.Request[api_gen.CallRequest]) (*connect.Response[api_gen.MakeTxResponse], error) {
+	for _, msg := range req.Msg.Msgs {
+		s.logger.Debug("MakeCallTx", zap.String("package", msg.PackagePath), zap.String("function", msg.Fnc), zap.Any("args", msg.Args))
+	}
+
+	cfg, msgs := convertCallRequest(req.Msg)
+	tx, err := s.client.MakeCallTx(*cfg, msgs...)
+	if err != nil {
+		return nil, getGrpcError(err)
+	}
+
+	txJSON, err := amino.MarshalJSON(tx)
+	if err != nil {
+		return nil, err
+	}
+	return connect.NewResponse(&api_gen.MakeTxResponse{TxJson: string(txJSON)}), nil
+}
+
+func (s *gnoNativeService) MakeSendTx(ctx context.Context, req *connect.Request[api_gen.SendRequest]) (*connect.Response[api_gen.MakeTxResponse], error) {
+	cfg, msgs := convertSendRequest(req.Msg)
+	tx, err := s.client.MakeSendTx(*cfg, msgs...)
+	if err != nil {
+		return nil, getGrpcError(err)
+	}
+
+	txJSON, err := amino.MarshalJSON(tx)
+	if err != nil {
+		return nil, err
+	}
+	return connect.NewResponse(&api_gen.MakeTxResponse{TxJson: string(txJSON)}), nil
+}
+
+func (s *gnoNativeService) MakeRunTx(ctx context.Context, req *connect.Request[api_gen.RunRequest]) (*connect.Response[api_gen.MakeTxResponse], error) {
+	cfg, msgs := convertRunRequest(req.Msg)
+	tx, err := s.client.MakeRunTx(*cfg, msgs...)
+	if err != nil {
+		return nil, getGrpcError(err)
+	}
+
+	txJSON, err := amino.MarshalJSON(tx)
+	if err != nil {
+		return nil, err
+	}
+	return connect.NewResponse(&api_gen.MakeTxResponse{TxJson: string(txJSON)}), nil
+}
+
+func (s *gnoNativeService) SignTx(ctx context.Context, req *connect.Request[api_gen.SignTxRequest]) (*connect.Response[api_gen.SignTxResponse], error) {
+	s.lock.RLock()
+	if s.activeAccount == nil {
+		s.lock.RUnlock()
+		return nil, api_gen.ErrCode_ErrNoActiveAccount
+	}
+	s.lock.RUnlock()
+
+	var tx std.Tx
+	if err := amino.UnmarshalJSON([]byte(req.Msg.TxJson), &tx); err != nil {
+		return nil, err
+	}
+
+	signedTx, err := s.client.SignTx(tx, req.Msg.AccountNumber, req.Msg.SequenceNumber)
```

Comment:
Before overwriting AccountNumber, shouldn't you test if it is empty? Because the field can be set when you make the transaction message if I'm not wrong.

Reply:
`AccountNumber` is different from the address. `AccountNumber` is an integer. The default is zero. `gnoclient` [sets `AccountNumber`](https://github.com/gnolang/gno/blob/14b9015875ee557008cb55b0bc454dcd326070d1/gno.land/pkg/gnoclient/client_txs.go#L308) if it is zero. Does that make sense?